aspnet/fundamentals/hosting.rst

```diff
@@ -0,0 +1,296 @@
+
+.. note:: Specifying ``Directory.GetCurrentDirectory`` as the content root will use the web project's root folder as the app's content root when the app is started from this folder (for example, calling ``dotnet run`` from the web project folder). This is the default used in Visual Studio and ``dotnet new`` templates.
+
+If the app should work with IIS, the ``UseIISIntegration`` method should be called as part of building the host. Note that this does not configure a *server*, like ``UseKestrel`` does. To use IIS with ASP.NET Core, you must specify both ``UseKestrel`` and ``UseIISIntegration``. Kestrel is designed to be run behind a proxy and should not be deployed directly facing the Internet. ``UseIISIntegration`` specifies IIS as the reverse proxy server.
+
+.. note:: ``UseKestrel`` and ``UseIISIntegration`` are very different actions. IIS is only used as a reverse proxy. ``UseKestrel`` creates the web server and hosts the code. ``UseIISIntegration`` specifies IIS as the reverse proxy server. It also examines environment variables used by IIS/IISExpress and makes decisions like which dynamic port use, which headers to set, etc. However, it doesn't deal with or create an ``IServer``.
+
+A minimal implementation of configuring a host (and an ASP.NET app) would include just a server and configuration of the app's request pipeline:
+
+.. code-block:: c#
+
+    var host = new WebHostBuilder()
+        .UseKestrel()
+        .Configure(app => 
+        {
+            app.Run(async (context) => await context.Response.WriteAsync("Hi!"));
+        })
+        .Build();
+
+    host.Run();
+
+.. note:: When setting up a host, you can provide ``Configure`` and ``ConfigureServices`` methods, instead of or in addition to specifying a ``Startup`` class (which must also define these methods - see :doc:`startup`).
+
+Configuring a Host
+------------------
+
+The ``WebHostBuilder`` provides useful methods for setting most of the available configuration values for the host, which can also be set directly using ``UseSetting`` and associated key. For example, to specify the application name:
+
+.. code-block:: c#
+
+    .UseSetting("applicationName", "MyApp")
+
+Host Configuration Values
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Application Name ``string``
+    Key: ``applicationName``. This configuration setting specifies the value that will be returned from ``IHostingEnvironment.ApplicationName``.
+
+Capture Startup Errors ``bool``
+    Key: ``captureStartupErrors``. Defaults to ``false``. When ``false``, errors during startup result in the host exiting. When ``true``, the host will capture any exceptions from the ``Startup`` class, and attempt to start the server anyway, displaying an error page (generic, or detailed, based on the Detailed Errors setting, below) in response to every request. Set using the ``CaptureStartupErrors`` method.
```

Comment:
consider:

When ``true``, the host will capture any exceptions from the ``Startup`` class ~~,~~ and attempt to start the server. It will display an error page for each request.


Is that true - if you get a recoverable exception in startup, every request for the life of the app will display an error message?


`throw new exception("Just kidding");`

will cause an error message for all requests?